### PR TITLE
Add support for 1-24 with feature flag

### PR DESCRIPTION
**Pull request by mitalipaygude**
*Issue #, if available:*
#3491

*Description of changes:*
Feature flag for 1.24 support.

*Testing (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/api/v1alpha1/zz_generated.deepcopy.go [hunk 1]
```diff
@@ -1,3 +1,4 @@
+//go:build !ignore_autogenerated
 // +build !ignore_autogenerated
```
Comment (vivek-koppuru) [inline pkg/api/v1alpha1/zz_generated.deepcopy.go:2]: This line should be removed right?

File: pkg/providers/snow/api/v1beta1/zz_generated.deepcopy.go [hunk 2]
```diff
@@ -1,3 +1,4 @@
+//go:build !ignore_autogenerated
 // +build !ignore_autogenerated
```
Comment (vivek-koppuru) [inline pkg/providers/snow/api/v1beta1/zz_generated.deepcopy.go:2]: same here

**Review by vivek-koppuru — APPROVED** [review]

File: pkg/api/v1alpha1/zz_generated.deepcopy.go (hunk 1 above)
Comment (mitalipaygude) [reply pkg/api/v1alpha1/zz_generated.deepcopy.go:2]: i think its a formatting change

File: pkg/providers/snow/api/v1beta1/zz_generated.deepcopy.go (hunk 2 above)
Comment (mitalipaygude) [reply pkg/providers/snow/api/v1beta1/zz_generated.deepcopy.go:2]: formatting change i think

**Review by vivek-koppuru — APPROVED** [review]

**Pull request merged**